account_analytic_moves_extend/model/account_move_line.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.multi
+    def create_analytic_lines(self):
+        acc_ana_line_obj = self.env['account.analytic.line']
+        for obj_line in self:
+            # Only create analytic line if the associated account is
+            # Expense or Income.
+            if obj_line.account_id.user_type_id.name or\
+                    obj_line.account_id.user_type_id.type in ('Income',
+                                                              'Expense'):
+                if obj_line.analytic_account_id:
```

Comment:
There's no analytic journal in v10. Moreover, you are overwriting super. I think you just need to check the account user type as you are doing, and then if it's Income or expenses call super.